modal/experimental.py

```diff
@@ -13,3 +32,161 @@ def get_local_input_concurrency():
     """Get the container's local input concurrency. Return 0 if the container is not running."""
 
     return _ContainerIOManager.get_input_concurrency()
+
+
+# START Experimental: Container Networking
+
+
+class _GroupedFunctionCall(typing.Generic[P, ReturnType, OriginalReturnType], _Object, type_prefix="gc"):
+    """Wrapper around _FunctionCall that allows for grouped functions to be spawned."""
+
+    def __init__(self, handles: List[FunctionCall]):
+        self.handles: List[FunctionCall] = handles
+
+    def get(self, *args: P.args, **kwargs: P.kwargs) -> List[ReturnType]:
+        """Get the result of a grouped function call."""
+        output: List[ReturnType] = []
+        for handle in self.handles:
+            output.append(handle.get())
+        return output
+
+    def get_gen(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
+        raise NotImplementedError("Grouped functions cannot be generators")
+
+    def get_call_graph(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
+        raise NotImplementedError("Grouped functions do not show call graph")
+
+    def cancel(
+        self,
+        terminate_containers: bool = False,
+    ) -> None:
+        for handle in self.handles:
+            handle.cancel(terminate_containers)
+
+
+class _GroupedFunction(typing.Generic[P, ReturnType, OriginalReturnType], _Object, type_prefix="gf"):
+    """Experimental wrapper around _Function that allows for containers to be spun up concurrently."""
+
+    def __init__(self, f: _Function, size: int):
+        self.f = synchronize_api(f)
+        self.size = size
+
+    def remote(self, *args: P.args, **kwargs: P.kwargs) -> List[ReturnType]:
+        """
+        Calls the function remotely, executing it with the given arguments and returning the execution's result.
+        """
+        handler = self.spawn(*args, **kwargs)
+        return handler.get()
+
+    def local(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
+        raise NotImplementedError("Grouped function cannot be run locally")
+
+    def spawn(self, *args: P.args, **kwargs: P.kwargs) -> _GroupedFunctionCall:
+        worker_handles: List[FunctionCall] = []
+        with modal.Queue.ephemeral() as q:
+            for i in range(self.size):
+                handle = self.f.spawn(*args, **kwargs, modal_rank=i, modal_size=self.size, modal_q=q)
+                worker_handles.append(handle)
+        handler: _GroupedFunctionCall = _GroupedFunctionCall(worker_handles)
+        return handler
+
+    def keep_warm(self, *args: P.args, **kwargs: P.kwargs) -> ReturnType:
```

Comment:
Could you override `__getattr__` to avoid needing to list all these methods as unimplemented? 

```python
def __getattr__(self, name):
        def unsupported_method(*args, **kwargs):
            raise NotImplementedError(f"Grouped function does not support the '{name}' method")
        return unsupported_method
```